standard/classes.md

```diff
@@ -5642,3 +5715,471 @@ An enumerable object may implement more interfaces than those specified above.
+- If there is a base record class type, the value of `base.Equals(other)` (a non-virtual call to `public virtual bool Equals(Base? other)`); otherwise
+the value of `EqualityContract == other.EqualityContract`.
+
+The record class type includes synthesized `==` and `!=` operators equivalent to operators declared as follows:
+
+```csharp
+public static bool operator==(R? left, R? right) =>
+    (object)left == right || (left?.Equals(right) ?? false);
+public static bool operator!=(R? left, R? right) => !(left == right);
+```
+
+The `Equals` method called by the `==` operator is the `Equals(R? other)` method specified above. The `!=` operator delegates to the `==` operator. It is an error if these operators are declared explicitly.
+
+If the record class type is derived from some base record class type, `Base`, the record class type includes a synthesized override equivalent to a method declared as follows:
+
+```csharp
+public sealed override bool Equals(Base? other);
+```
+
+It is an error if the override is declared explicitly. It is an error if the method doesn't override a method with the same signature in record class type `Base` (for example, if the method is missing in the `Base`, or is sealed, or is not virtual). The synthesized override returns `Equals((object?)other)`.
+
+The record class type includes a synthesized override equivalent to a method declared as follows:
+
+```csharp
+public override bool Equals(object? obj);
+```
+
+It is an error if the override is declared explicitly. It is an error if the method doesn't override `object.Equals(object? obj)` (for example, due to shadowing in intermediate base types). The synthesized override returns `Equals(other as R)` where `R` is the record class type.
+
+The record class type includes a synthesized override equivalent to a method declared as follows:
+
+```csharp
+public override int GetHashCode();
+```
+
+The method may be declared explicitly. It is an error if the explicit declaration doesn't allow overriding it in a derived type and the record class type is not `sealed`. It is an error if either the synthesized, or the explicitly declared, method doesn't override `object.GetHashCode()` (for example, due to shadowing in intermediate base types).
+
+A warning shall be issued if one of `Equals(R?)` and `GetHashCode()` is explicitly declared, but the other is not.
+
+The synthesized override of `GetHashCode()` returns an `int` result of combining the following values:
+
+- For each instance field `fieldN` in the record class type that is not inherited, the value of `System.Collections.Generic.EqualityComparer<TN>.Default.GetHashCode(fieldN)` where `TN` is the field type, and
+- If there is a base record class type, the value of `base.GetHashCode()`; otherwise
+the value of `System.Collections.Generic.EqualityComparer<System.Type>.Default.GetHashCode(EqualityContract)`.
```

Comment:
```suggestion
- If there is a base record class type, the value of `base.GetHashCode()`; otherwise the value of `System.Collections.Generic.EqualityComparer<System.Type>.Default.GetHashCode(EqualityContract)`.
```